packages/grid/_modules_/grid/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -327,6 +327,13 @@ export const useGridColumnResize = (apiRef: GridApiRef) => {
     };
   }, [apiRef, handleTouchStart, stopListening]);
 
+  React.useEffect(() => {
+    // TODO properly handle listener cleanup on grid unmount
```

Comment:
Run tests without this effect

Reply:
If I remove the effect, I have the same error as described here :
https://github.com/mui-org/material-ui-x/pull/1989#discussion_r661220534

Reply:
@dtassone do you think we better merge this PR as it is or improve the behavior of `useGridColumnResize` to avoid this non-conditional cleanup on unmount ?

Reply:
IMO we can do it in a separate PR, and we could capture the refactoring in a separate issue.